--- git status ---
On branch Lift-Needle-Task
Your branch is ahead of 'origin/Lift-Needle-Task' by 3 commits.
  (use "git push" to publish your local commits)

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   source/extensions/orbit.surgical.assets/orbit/surgical/assets/mops.py
	modified:   source/extensions/orbit.surgical.tasks/orbit/surgical/tasks/surgical/lift_needle/config/mops/joint_pos_env_cfg.py
	modified:   source/extensions/orbit.surgical.tasks/orbit/surgical/tasks/surgical/lift_needle/lift_env_cfg.py
	modified:   source/standalone/workflows/rsl_rl/play.py
	modified:   source/standalone/workflows/rsl_rl/train.py

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	rename_isaac_lab.py

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/source/extensions/orbit.surgical.assets/orbit/surgical/assets/mops.py b/source/extensions/orbit.surgical.assets/orbit/surgical/assets/mops.py
index a60b3f3..2622d98 100644
--- a/source/extensions/orbit.surgical.assets/orbit/surgical/assets/mops.py
+++ b/source/extensions/orbit.surgical.assets/orbit/surgical/assets/mops.py
@@ -20,7 +20,7 @@ import torch
 ##
 
 # Given real-world position
-world_to_base_pos = [0, -0.0765, 0.92]
+world_to_base_pos = [0, -0.0765, 0.92 - 0.2]
 
 # Given real-world RPY (roll, pitch, yaw) in radians
 world_to_base_rpy = torch.tensor([-1.1224, 0.3337, -2.6655]) 
@@ -93,7 +93,7 @@ MOPS_CFG = ArticulationCfg(
             joint_names_expr=["lnd_tool_gripper.*"],
             effort_limit=1.0,
             velocity_limit=0.5, 
-            stiffness=500,  
+            stiffness=50,  
             damping=0.1,  
         ),
     },
diff --git a/source/extensions/orbit.surgical.tasks/orbit/surgical/tasks/surgical/lift_needle/config/mops/joint_pos_env_cfg.py b/source/extensions/orbit.surgical.tasks/orbit/surgical/tasks/surgical/lift_needle/config/mops/joint_pos_env_cfg.py
index f688873..702d07c 100644
--- a/source/extensions/orbit.surgical.tasks/orbit/surgical/tasks/surgical/lift_needle/config/mops/joint_pos_env_cfg.py
+++ b/source/extensions/orbit.surgical.tasks/orbit/surgical/tasks/surgical/lift_needle/config/mops/joint_pos_env_cfg.py
@@ -46,7 +46,7 @@ class NeedleLiftEnvCfg(LiftEnvCfg):
                 "lnd_tool_pitch_joint",
                 "lnd_tool_yaw_joint",
             ],
-            scale=0.5,
+            scale=0.2,
             use_default_offset=True,
         )
         self.actions.finger_joint_pos = mdp.BinaryJointPositionActionCfg(
diff --git a/source/extensions/orbit.surgical.tasks/orbit/surgical/tasks/surgical/lift_needle/lift_env_cfg.py b/source/extensions/orbit.surgical.tasks/orbit/surgical/tasks/surgical/lift_needle/lift_env_cfg.py
index 7fb74fc..ffd9621 100644
--- a/source/extensions/orbit.surgical.tasks/orbit/surgical/tasks/surgical/lift_needle/lift_env_cfg.py
+++ b/source/extensions/orbit.surgical.tasks/orbit/surgical/tasks/surgical/lift_needle/lift_env_cfg.py
@@ -72,18 +72,20 @@ class CommandsCfg:
     object_pose = mdp.UniformPoseCommandCfg(
         asset_name="robot",
         body_name=MISSING,  # will be set by agent env cfg
-        resampling_time_range=(5.0, 5.0),
+        resampling_time_range=(1.0, 1.0),
         debug_vis=False,
         ranges=mdp.UniformPoseCommandCfg.Ranges(
-            pos_x=(-0.8, -0.7),
-            pos_y=(0.6, 0.8),
-            pos_z=(-0.1, 0.1),
-            roll=(-np.pi / 2, -np.pi / 2),
-            pitch=(-np.pi / 4, np.pi / 4),
-            yaw=(0.0, 0.0),
+            pos_x=(-0.36, -0.33),
+            pos_y=(0.7, 0.73),
+            pos_z=(-0.2, -0.17),
+            roll=(-1.12899527 + np.pi, -1.12899527 + np.pi),
+            pitch=(-0.29088276, -0.29088276),
+            yaw=(2.63952309, 2.63952309),
         ),
     )
-
+    # set the scale of the visualization markers to (0.01, 0.01, 0.01)
+    object_pose.goal_pose_visualizer_cfg.markers["frame"].scale = (0.01, 0.01, 0.01)
+    object_pose.current_pose_visualizer_cfg.markers["frame"].scale = (0.01, 0.01, 0.01)
 
 @configclass
 class ActionsCfg:
@@ -126,7 +128,7 @@ class EventCfg:
         func=mdp.reset_root_state_uniform,
         mode="reset",
         params={
-            "pose_range": {"x": (-0.03, 0.1), "y": (-0.03, 0.5), "z": (0.0, 0.0)},
+            "pose_range": {"x": (-0.03, 0.03), "y": (0.43, 0.46), "z": (0.0, 0.0)},
             "velocity_range": {},
             "asset_cfg": SceneEntityCfg("object", body_names="Object"),
         },
@@ -220,7 +222,7 @@ class LiftEnvCfg(ManagerBasedRLEnvCfg):
         # general settings
         self.decimation = 4
         self.sim.render_interval = self.decimation
-        self.episode_length_s = 5.0
+        self.episode_length_s = 2.0
         # simulation settings
         self.sim.dt = 1.0 / 200.0
         self.viewer.eye = (4, -0.6, 0.3)
diff --git a/source/standalone/workflows/rsl_rl/play.py b/source/standalone/workflows/rsl_rl/play.py
index fd017e2..3c139e8 100644
--- a/source/standalone/workflows/rsl_rl/play.py
+++ b/source/standalone/workflows/rsl_rl/play.py
@@ -22,7 +22,6 @@ parser.add_argument(
 parser.add_argument("--num_envs", type=int, default=None, help="Number of environments to simulate.")
 parser.add_argument("--task", type=str, default=None, help="Name of the task.")
 parser.add_argument("--seed", type=int, default=None, help="Seed used for the environment")
-parser.add_argument("--gpu", type=bool, default=True, help="Whether to use GPU/CPU pipeline.")
 # append RSL-RL cli arguments
 cli_args.add_rsl_rl_args(parser)
 # append AppLauncher cli args
@@ -57,7 +56,7 @@ def main():
     """Play with RSL-RL agent."""
     # parse configuration
     env_cfg = parse_env_cfg(
-        args_cli.task, use_gpu=args_cli.gpu, num_envs=args_cli.num_envs, use_fabric=not args_cli.disable_fabric
+        args_cli.task, num_envs=args_cli.num_envs, use_fabric=not args_cli.disable_fabric
     )
     agent_cfg: RslRlOnPolicyRunnerCfg = cli_args.parse_rsl_rl_cfg(args_cli.task, args_cli)
 
diff --git a/source/standalone/workflows/rsl_rl/train.py b/source/standalone/workflows/rsl_rl/train.py
index 9d517f9..b0b0feb 100644
--- a/source/standalone/workflows/rsl_rl/train.py
+++ b/source/standalone/workflows/rsl_rl/train.py
@@ -69,7 +69,7 @@ def main():
     """Train with RSL-RL agent."""
     # parse configuration
     env_cfg: ManagerBasedRLEnvCfg = parse_env_cfg(
-        args_cli.task, use_gpu=args_cli.gpu, num_envs=args_cli.num_envs, use_fabric=not args_cli.disable_fabric
+        args_cli.task, num_envs=args_cli.num_envs, use_fabric=not args_cli.disable_fabric
     )
     agent_cfg: RslRlOnPolicyRunnerCfg = cli_args.parse_rsl_rl_cfg(args_cli.task, args_cli)
 