--- git status ---
On branch Lift-Needle-Task
Your branch is ahead of 'origin/Lift-Needle-Task' by 3 commits.
  (use "git push" to publish your local commits)

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   source/extensions/orbit.surgical.assets/orbit/surgical/assets/mops.py
	modified:   source/extensions/orbit.surgical.tasks/orbit/surgical/tasks/surgical/lift_needle/config/mops/agents/rsl_rl_cfg.py
	modified:   source/extensions/orbit.surgical.tasks/orbit/surgical/tasks/surgical/lift_needle/config/mops/joint_pos_env_cfg.py
	modified:   source/extensions/orbit.surgical.tasks/orbit/surgical/tasks/surgical/lift_needle/lift_env_cfg.py
	modified:   source/standalone/workflows/rsl_rl/play.py
	modified:   source/standalone/workflows/rsl_rl/train.py

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	rename_isaac_lab.py

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/source/extensions/orbit.surgical.assets/orbit/surgical/assets/mops.py b/source/extensions/orbit.surgical.assets/orbit/surgical/assets/mops.py
index a60b3f3..2622d98 100644
--- a/source/extensions/orbit.surgical.assets/orbit/surgical/assets/mops.py
+++ b/source/extensions/orbit.surgical.assets/orbit/surgical/assets/mops.py
@@ -20,7 +20,7 @@ import torch
 ##
 
 # Given real-world position
-world_to_base_pos = [0, -0.0765, 0.92]
+world_to_base_pos = [0, -0.0765, 0.92 - 0.2]
 
 # Given real-world RPY (roll, pitch, yaw) in radians
 world_to_base_rpy = torch.tensor([-1.1224, 0.3337, -2.6655]) 
@@ -93,7 +93,7 @@ MOPS_CFG = ArticulationCfg(
             joint_names_expr=["lnd_tool_gripper.*"],
             effort_limit=1.0,
             velocity_limit=0.5, 
-            stiffness=500,  
+            stiffness=50,  
             damping=0.1,  
         ),
     },
diff --git a/source/extensions/orbit.surgical.tasks/orbit/surgical/tasks/surgical/lift_needle/config/mops/agents/rsl_rl_cfg.py b/source/extensions/orbit.surgical.tasks/orbit/surgical/tasks/surgical/lift_needle/config/mops/agents/rsl_rl_cfg.py
index 3bef725..2a53801 100644
--- a/source/extensions/orbit.surgical.tasks/orbit/surgical/tasks/surgical/lift_needle/config/mops/agents/rsl_rl_cfg.py
+++ b/source/extensions/orbit.surgical.tasks/orbit/surgical/tasks/surgical/lift_needle/config/mops/agents/rsl_rl_cfg.py
@@ -14,8 +14,8 @@ from isaaclab_rl.rsl_rl import (
 @configclass
 class LiftNeedlePPORunnerCfg(RslRlOnPolicyRunnerCfg):
     num_steps_per_env = 24
-    max_iterations = 3000
-    save_interval = 100
+    max_iterations = 5000
+    save_interval = 300
     experiment_name = "mops_needle_lift"
     empirical_normalization = False
     policy = RslRlPpoActorCriticCfg(
diff --git a/source/extensions/orbit.surgical.tasks/orbit/surgical/tasks/surgical/lift_needle/config/mops/joint_pos_env_cfg.py b/source/extensions/orbit.surgical.tasks/orbit/surgical/tasks/surgical/lift_needle/config/mops/joint_pos_env_cfg.py
index f688873..702d07c 100644
--- a/source/extensions/orbit.surgical.tasks/orbit/surgical/tasks/surgical/lift_needle/config/mops/joint_pos_env_cfg.py
+++ b/source/extensions/orbit.surgical.tasks/orbit/surgical/tasks/surgical/lift_needle/config/mops/joint_pos_env_cfg.py
@@ -46,7 +46,7 @@ class NeedleLiftEnvCfg(LiftEnvCfg):
                 "lnd_tool_pitch_joint",
                 "lnd_tool_yaw_joint",
             ],
-            scale=0.5,
+            scale=0.2,
             use_default_offset=True,
         )
         self.actions.finger_joint_pos = mdp.BinaryJointPositionActionCfg(
diff --git a/source/extensions/orbit.surgical.tasks/orbit/surgical/tasks/surgical/lift_needle/lift_env_cfg.py b/source/extensions/orbit.surgical.tasks/orbit/surgical/tasks/surgical/lift_needle/lift_env_cfg.py
index 7fb74fc..06949a3 100644
--- a/source/extensions/orbit.surgical.tasks/orbit/surgical/tasks/surgical/lift_needle/lift_env_cfg.py
+++ b/source/extensions/orbit.surgical.tasks/orbit/surgical/tasks/surgical/lift_needle/lift_env_cfg.py
@@ -72,18 +72,20 @@ class CommandsCfg:
     object_pose = mdp.UniformPoseCommandCfg(
         asset_name="robot",
         body_name=MISSING,  # will be set by agent env cfg
-        resampling_time_range=(5.0, 5.0),
-        debug_vis=False,
+        resampling_time_range=(1.0, 1.0),
+        debug_vis=True,
         ranges=mdp.UniformPoseCommandCfg.Ranges(
-            pos_x=(-0.8, -0.7),
-            pos_y=(0.6, 0.8),
-            pos_z=(-0.1, 0.1),
-            roll=(-np.pi / 2, -np.pi / 2),
-            pitch=(-np.pi / 4, np.pi / 4),
-            yaw=(0.0, 0.0),
+            pos_x=(-0.53, -0.5),
+            pos_y=(0.4, 0.43),
+            pos_z=(-0.13, -0.1),
+            roll=(-1.12899527 + np.pi, -1.12899527 + np.pi),
+            pitch=(-0.29088276, -0.29088276),
+            yaw=(2.63952309, 2.63952309),
         ),
     )
-
+    # set the scale of the visualization markers to (0.01, 0.01, 0.01)
+    object_pose.goal_pose_visualizer_cfg.markers["frame"].scale = (0.01, 0.01, 0.01)
+    object_pose.current_pose_visualizer_cfg.markers["frame"].scale = (0.01, 0.01, 0.01)
 
 @configclass
 class ActionsCfg:
@@ -126,7 +128,7 @@ class EventCfg:
         func=mdp.reset_root_state_uniform,
         mode="reset",
         params={
-            "pose_range": {"x": (-0.03, 0.1), "y": (-0.03, 0.5), "z": (0.0, 0.0)},
+            "pose_range": {"x": (-0.03, 0.03), "y": (0.43, 0.46), "z": (0.0, 0.0)},
             "velocity_range": {},
             "asset_cfg": SceneEntityCfg("object", body_names="Object"),
         },
@@ -139,18 +141,18 @@ class RewardsCfg:
 
     reaching_object = RewTerm(func=mdp.object_ee_distance, params={"std": 0.1}, weight=1.0)
 
-    lifting_object = RewTerm(func=mdp.object_is_lifted, params={"minimal_height": 0.02}, weight=15.0)
+    lifting_object = RewTerm(func=mdp.object_is_lifted, params={"minimal_height": 0.02}, weight=10.0) 
 
     object_goal_tracking = RewTerm(
         func=mdp.object_goal_distance,
         params={"std": 0.3, "minimal_height": 0.02, "command_name": "object_pose"},
-        weight=16.0,
+        weight=20.0,  
     )
 
     object_goal_tracking_fine_grained = RewTerm(
         func=mdp.object_goal_distance,
         params={"std": 0.05, "minimal_height": 0.02, "command_name": "object_pose"},
-        weight=5.0,
+        weight=10.0,  
     )
 
     # action penalty
@@ -175,8 +177,8 @@ class TerminationsCfg:
         func=mdp.root_height_below_minimum, params={"minimum_height": -0.05, "asset_cfg": SceneEntityCfg("object")}
     )
 
-    # Terminate when the object is lifted and reaches a goal height
-    object_lifted = DoneTerm(func=mdp.object_reached_goal, params={"threshold": 0.1})
+    # Terminate when the object is lifted and reaches a goal position
+    object_lifted = DoneTerm(func=mdp.object_reached_goal, params={"threshold": 0.05})
 
 
 @configclass
@@ -204,7 +206,7 @@ class LiftEnvCfg(ManagerBasedRLEnvCfg):
     """Configuration for the lifting environment."""
 
     # Scene settings
-    scene: ObjectTableSceneCfg = ObjectTableSceneCfg(num_envs=512, env_spacing=2.5)
+    scene: ObjectTableSceneCfg = ObjectTableSceneCfg(num_envs=1024, env_spacing=2.5)
     # Basic settings
     observations: ObservationsCfg = ObservationsCfg()
     actions: ActionsCfg = ActionsCfg()
@@ -220,7 +222,7 @@ class LiftEnvCfg(ManagerBasedRLEnvCfg):
         # general settings
         self.decimation = 4
         self.sim.render_interval = self.decimation
-        self.episode_length_s = 5.0
+        self.episode_length_s = 2.0
         # simulation settings
         self.sim.dt = 1.0 / 200.0
         self.viewer.eye = (4, -0.6, 0.3)
diff --git a/source/standalone/workflows/rsl_rl/play.py b/source/standalone/workflows/rsl_rl/play.py
index fd017e2..3c139e8 100644
--- a/source/standalone/workflows/rsl_rl/play.py
+++ b/source/standalone/workflows/rsl_rl/play.py
@@ -22,7 +22,6 @@ parser.add_argument(
 parser.add_argument("--num_envs", type=int, default=None, help="Number of environments to simulate.")
 parser.add_argument("--task", type=str, default=None, help="Name of the task.")
 parser.add_argument("--seed", type=int, default=None, help="Seed used for the environment")
-parser.add_argument("--gpu", type=bool, default=True, help="Whether to use GPU/CPU pipeline.")
 # append RSL-RL cli arguments
 cli_args.add_rsl_rl_args(parser)
 # append AppLauncher cli args
@@ -57,7 +56,7 @@ def main():
     """Play with RSL-RL agent."""
     # parse configuration
     env_cfg = parse_env_cfg(
-        args_cli.task, use_gpu=args_cli.gpu, num_envs=args_cli.num_envs, use_fabric=not args_cli.disable_fabric
+        args_cli.task, num_envs=args_cli.num_envs, use_fabric=not args_cli.disable_fabric
     )
     agent_cfg: RslRlOnPolicyRunnerCfg = cli_args.parse_rsl_rl_cfg(args_cli.task, args_cli)
 
diff --git a/source/standalone/workflows/rsl_rl/train.py b/source/standalone/workflows/rsl_rl/train.py
index 9d517f9..b0b0feb 100644
--- a/source/standalone/workflows/rsl_rl/train.py
+++ b/source/standalone/workflows/rsl_rl/train.py
@@ -69,7 +69,7 @@ def main():
     """Train with RSL-RL agent."""
     # parse configuration
     env_cfg: ManagerBasedRLEnvCfg = parse_env_cfg(
-        args_cli.task, use_gpu=args_cli.gpu, num_envs=args_cli.num_envs, use_fabric=not args_cli.disable_fabric
+        args_cli.task, num_envs=args_cli.num_envs, use_fabric=not args_cli.disable_fabric
     )
     agent_cfg: RslRlOnPolicyRunnerCfg = cli_args.parse_rsl_rl_cfg(args_cli.task, args_cli)
 